scripts/system/libraries/entitySelectionTool.js

```diff
@@ -1084,6 +1083,35 @@ SelectionDisplay = (function() {
         return tool;
     }
 
+    // toolHandle:  The overlayID associated with the tool
+    // that correlates to the tool you wish to query.
+    // @return: bool - Indicates if the activeTool is that queried.
+    function isActiveTool(toolHandle) {
+        var wantDebug = true;
+        if ( ! activeTool || ! toolHandle ){
+            if(wantDebug){
+                if ( activeTool ){
+                    print("WARNING: entitySelectionTool.isActiveTool - Encountered invalid toolHandle: " + toolHandle );
+                }
+            }
+            return false;
+        }
+        if ( ! grabberTools.hasOwnProperty( toolHandle ) ) {
+            if(wantDebug){
+                print("WARNING: entitySelectionTool.isActiveTool - Encountered unknown grabberToolHandle: " + toolHandle );
+            }
+            return false;
```

Comment:
I'm not sure how the <code>return false</code> here is misleading.  All active tools are expected to have been registered via the grabberTools array.  If the toolHandle provided isn't valid (isn't known and valid with respect to grabberTools), it can't be the active tool. If I let it fall through to hit the test, then the invalid handle will be added to the array as a key with a value of undefined which is something that's undesired.